### ISSUE TITLE
Sync.pull returns (_, pull_error) result, but there is no pp_pull_error

### ISSUE THREAD

**Issue by hannesm** [mirage/irmin#755]
thanks for #684, I finally found the time to attempt to upgrade to a more recent irmin-master.

while doing so, the signature of Sync.pull changed:
```
-     (unit, [ fetch_error | Merge.conflict ]) result Lwt.t
+    (status, pull_error) result Lwt.t
```

now in #683 (not merged) a `pp_pull_error` was briefly introduced, but the current API does not have a `pp_pull_error` (but it has a `pp_push_error`). It would be great to supply potential users with such a functionality?

**Comment by hannesm** [comment]
and while at it, a `pp_status` may be beneficial as well ;)

**Comment by hannesm** [comment]
turns out there's `S.Store.Status` which defines `t` and `pp`, but the `S.SYNC_STORE.status` is slightly different... (and the `pp` doesn't contain any variance annotations)...

**Comment by hannesm** [comment]
in addition, your `src/irmin-git/irmin_git.ml` contains some debug messages which are not very useful:
```OCaml
    let result refs =
      (* FIXME: need pp_result *)
      Log.debug (fun f ->
          f "fetch result: XXX" (* Git.Sync.Result.pp_fetch r *) );
      let key =
        try Some (G.Reference.Map.find reference refs) with Not_found -> None
      in
```

**Comment by CraigFe** [comment]
Thanks for the reports. Looks like there already was a `pp_pull_error`, but it just wasn't being exposed.

This is partly resolved by https://github.com/mirage/irmin/pull/789 (which doesn't do anything about the incomplete debug message, yet); please let me know if it's missing something.

**Comment by hannesm** [comment]
@CraigFe thanks, that looks like a good solution to me! :)

**Comment by samoht** [comment]
Fixed by #789
